commons/src/main/java/org/infinispan/commons/api/functional/Param.java

```diff
@@ -44,6 +44,29 @@
    P get();
 
    /**
+    * Should this paramater go over the wire?
+    */
+   SerializeMethod serializeMethod();
+
+   /**
+    * The way how this object should be serialized;
+    */
+   enum SerializeMethod {
```

Comment:
Also, whether the param itself should be sent over the wire or not is not down the parameter itself, but more to the command being executed. The command itself must decide whether it includes certain params inside its externalizer payload. Params themselves do not live on their own.